aws/resource_aws_glue_catalog_crawler_test.go

```diff
@@ -0,0 +1,276 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"testing"
+)
+
+func TestAccAWSGlueCrawler_basic(t *testing.T) {
+	const name = "aws_glue_catalog_crawler.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigBasic,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test-basic"),
+					resource.TestCheckResourceAttr(name, "name", "test-basic"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "AWSGlueServiceRole-tf"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSGlueCrawler_jdbcCrawler(t *testing.T) {
+	const name = "aws_glue_catalog_crawler.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigJdbc,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test-jdbc"),
+					resource.TestCheckResourceAttr(name, "name", "test-jdbc"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "AWSGlueServiceRoleDefault"),
+					resource.TestCheckResourceAttr(name, "jdbc_target.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSGlueCrawler_customCrawlers(t *testing.T) {
+	const name = "aws_glue_catalog_crawler.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigCustomClassifiers,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test_custom"),
+					resource.TestCheckResourceAttr(name, "name", "test_custom"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "tf-glue-service-role"),
+					resource.TestCheckResourceAttr(name, "table_prefix", "table_prefix"),
+					resource.TestCheckResourceAttr(name, "schema_change_policy.0.delete_behavior", "DELETE_FROM_DATABASE"),
+					resource.TestCheckResourceAttr(name, "schema_change_policy.0.update_behavior", "UPDATE_IN_DATABASE"),
+					resource.TestCheckResourceAttr(name, "s3_target.#", "2"),
+					resource.TestCheckResourceAttr(name, "s3_target.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func checkGlueCatalogCrawlerExists(name string, crawlerName string) resource.TestCheckFunc {
```

Comment:
The `crawlerName` parameter here is extraneous, its available in `rs.Primary.ID`